### ISSUE TITLE
Disable Auto-play in playlists

### ISSUE THREAD

**Issue by insin** [insin/control-panel-for-youtube#3]
When watching a video in a playlist, YouTube hides the Auto-play control in the player, and doesn't respect it if you had turned it off

On desktop, we can force playlist Auto-play off like so: 

```js
// Disable Autoplay in playlists
let $playlistManager = document.querySelector('yt-playlist-manager')
Object.defineProperty($playlistManager, 'canAutoAdvance_', {
  get() { return !config.disablePlaylistAutoplay },
  set() {},
  enumerable: true,
  configurable: true,
})
```

We can either un-hide the Auto-play control in the video toolbar and use its state in the code above, or add our own control to the playlist UI next to the existing repeat and shuffle controls it have.

I haven't found a similar way to control playlist Auto-play on mobile yet, so we might need to do it the hard way.

> [!NOTE]
> This requires access to DOM expandos, so we'd need to inject `content.js` as a <script> instead

**Comment by insin** [comment]
I don't think this is necessary given the purpose of a playlist, but we could use the information above in future if anything similar is needed

**Comment by dtruebin** [comment]
I used to use a bunch of different extensions that tweak various aspects of YouTube.
Having discovered this one (thank you!), I ended up removing almost all the rest of them, but this feature here is the one I truly miss.

My usage scenario:
1. add a bunch of videos from my subscription feed to the Queue (which is, in essence, a playlist)
2. watch the first video
3. _at the end, pause_
4. check out the additional info in the video description, comments, etc.
5. manually advance to the next video
6. repeat from step 2 until the end of the queue is reached

Doing step 3 manually is tedious as I have to remember to pause the video before it ends. This is worsened by the fact that end-of-video experience is inconsistent: some videos have an obvious outro, and some end "abruptly". So, with no auto-pause, I often end up having to pause the video long before its meaningful part ends -- just because I want to avoid the frustration of getting auto-switched to the next video and having to manually switch back (at which point the thing I _just watched_ begins playing _again_ argh).
Automating step 3 would address this shortcoming of YouTube UX 🙏 

Until then, an additional dedicated extension (e.g., see [Chrome Web Store search](https://chromewebstore.google.com/search/youtube%20playlist%20autoplay?hl=en-US&utm_source=ext_sidebar)) will have do the trick.

**Comment by realAzazello** [comment]
Which previous extension did you use that satisfied your scenario?

As an alternative you could try a userscript - try this search on GF: https://greasyfork.org/scripts?per_page=100&new=1&q=youtube+playlist+Autoplay&sort=updated

There's a recent script that sounds like it does what you want: https://greasyfork.org/scripts/546046. (I've never tried it.) If it works, maybe this repo's i.e. CPfYT owner could incorporate the code, assuming the license is flexible.

Not what your scenario is, but check out a related suggestion @ https://github.com/insin/control-panel-for-youtube/issues/184#issuecomment-3707289758
